internal/daemon/controller/handlers/targets/target_service.go

```diff
@@ -234,87 +234,115 @@ func (s Service) ListTargets(ctx context.Context, req *pbs.ListTargetsRequest) (
 		return &pbs.ListTargetsResponse{}, nil
 	}
 
-	repo, err := s.repoFn(target.WithPermissions(userPerms))
+	pageSize := int(s.maxPageSize)
+	// Use the requested page size only if it is smaller than
+	// the configured max.
+	if req.GetPageSize() != 0 && uint(req.GetPageSize()) < s.maxPageSize {
+		pageSize = int(req.GetPageSize())
+	}
+	filter, err := handlers.NewFilter(ctx, req.GetFilter())
 	if err != nil {
 		return nil, err
 	}
-	filter, err := handlers.NewFilter(ctx, req.GetFilter())
+
+	repo, err := s.repoFn(target.WithPermissions(userPerms))
 	if err != nil {
 		return nil, err
 	}
-	listTargetsFn := func(prevPageLast target.Target, refreshToken *pbs.ListRefreshToken, limit int) ([]target.Target, error) {
-		opts := []target.Option{
-			target.WithLimit(limit),
+	filterItemFn := func(item target.Target) (bool, error) {
```

Comment:
This function is used in the domain to allow filtering of items based on permissions and the user supplied filter. The alternative we considered is to write some logic for converting the application level filtering to a domain level filter (that would work across all domains), but this would certainly mean we would miss the November release. We could come back to refactor this at a later point if we miss the November release.